File: Python/7py 3/solve_as.py
"""
|--------------------------------------------------------------------------
| Alphabet symmetry
|--------------------------------------------------------------------------
|
"""

def solve(arr):
   abj = "abcdefghijklmnopqrstuvwxyzabcdefghijklmnopqrstuvwxyz"
   res = []
   for i, v in enumerate(arr):
      count = 0
      for i2, v2 in enumerate(v):
         if v2.lower() == abj[i2]: count += 1
      res.append(count)
   return res


def solve(arr):
   return [ sum(c == chr(97+i) for i,c in enumerate(v[:26].lower())) for v in arr ]


def solve(arr):
   return [sum(a==b for a, b in zip(v.lower(), 'abcdefghijklmnopqrstuvwxyz')) for v in arr]


def solve(arr):
   return [sum(i == ord(c) - ord('A') for i, c in enumerate(s.upper())) for s in arr]
   

print(solve(["abode","ABc","xyzD"]),[4,3,1])
print(solve(["abide","ABc","xyz"]),[4,3,0])
print(solve(["IAMDEFANDJKL","thedefgh","xyzDEFghijabc"]),[6,5,7])
print(solve(["encode","abc","xyzD","ABmD"]),[1, 3, 1, 3])

File: PHP/7php 3/solve_as.php
<?php
/*
|--------------------------------------------------------------------------
| Alphabet symmetry
|--------------------------------------------------------------------------
|
*/

function solve($arr) {
   $abj = "abcdefghijklmnopqrstuvwxyz";
   $res = [];
   for ($i = 0; $i < count($arr); $i++) {
      $count = 0;
      for ($j = 0; $j < strlen($arr[$i]); $j++) {
         if (strtolower($arr[$i][$j]) == $abj[$j]) $count += 1;
      }
      array_push($res, $count);
   }
   return $res;
}


function solve($arr) {
   return array_map(function($n) {
      return count(array_intersect_assoc(range('a', 'z'), str_split(strtolower($n))));
   }, $arr);
}


function solve($arr) {
   $alp = range('a', 'z');
   $res = [];
   foreach($arr as $string) {
      array_push($res, count(array_intersect_assoc(str_split(strtolower($string)), $alp)));
   }
   return $res;
}


print_r(solve(["abode", "ABc", "xyzD"]));
print_r(solve(["abide", "ABc", "xyz"]));
print_r(solve(["IAMDEFANDJKL", "thedefgh", "xyzDEFghijabc"]));
print_r(solve(["encode", "abc", "xyzD", "ABmD"]));